allennlp/modules/conditional_random_field.py

```diff
@@ -0,0 +1,170 @@
+"""
+Conditional random field
+"""
+
+import torch
+
+def log_sum_exp(x: torch.Tensor) -> torch.Tensor:  # pylint: disable=invalid-name
```

Comment:
I just merged a PR that put the version of this function from the `coref-wip` branch into master, because I needed it for something else I was doing.  You can just use that one - it has a little bit more functionality than this one.

Reply:
👍